argilla-server/tests/unit/api/handlers/v1/datasets/records/records_bulk/test_create_dataset_records_bulk.py

```diff
@@ -214,3 +217,198 @@ async def test_create_dataset_records(
+            json={
+                "items": [
+                    {
+                        "fields": {
+                            "image": f"https://argilla.io/{'a' * 2038}.jpeg",
+                        },
+                    },
+                ],
+            },
+        )
+
+        assert response.status_code == 422
+        assert response.json() == {
+            "detail": f"record at position 0 is not valid because image field 'image' value is exceeding the maximum length of 2038 characters for Web URLs",
+        }
+
+        assert (await db.execute(select(func.count(Record.id)))).scalar_one() == 0
+
+    async def test_create_dataset_records_bulk_with_image_field_data_url_using_invalid_mime_type(
+        self, db: AsyncSession, async_client: AsyncClient, owner_auth_header: dict
+    ):
+        dataset = await DatasetFactory.create(status=DatasetStatus.ready)
+
+        await ImageFieldFactory.create(name="image", dataset=dataset)
+        await LabelSelectionQuestionFactory.create(dataset=dataset)
+
+        response = await async_client.post(
+            self.url(dataset.id),
+            headers=owner_auth_header,
+            json={
+                "items": [
+                    {
+                        "fields": {
+                            "image": "data:image/invalid;base64,UklGRhgCAABXRUJQVlA4WAoAAAAIAAAAHwAAFwA",
+                        },
+                    },
+                ],
+            },
+        )
+
+        assert response.status_code == 422
+        assert response.json() == {
+            "detail": f"record at position 0 is not valid because image field 'image' value is using an unsupported MIME type, supported MIME types are: ['image/avif', 'image/gif', 'image/ico', 'image/jpeg', 'image/jpg', 'image/png', 'image/svg', 'image/webp']",
+        }
+
+        assert (await db.execute(select(func.count(Record.id)))).scalar_one() == 0
+
+    async def test_create_dataset_records_bulk_with_image_field_data_url_exceeding_maximum_length(
+        self, db: AsyncSession, async_client: AsyncClient, owner_auth_header: dict
+    ):
+        dataset = await DatasetFactory.create(status=DatasetStatus.ready)
+
+        await ImageFieldFactory.create(name="image", dataset=dataset)
+        await LabelSelectionQuestionFactory.create(dataset=dataset)
+
+        response = await async_client.post(
+            self.url(dataset.id),
+            headers=owner_auth_header,
+            json={
+                "items": [
+                    {
+                        "fields": {
+                            "image": f"data:image/jpeg;base64,{'a' * 5_000_000}",
+                        },
+                    },
+                ],
+            },
+        )
+
+        assert response.status_code == 422
+        assert response.json() == {
+            "detail": f"record at position 0 is not valid because image field 'image' value is exceeding the maximum length of 5000000 characters for Data URLs",
```

Comment:
@burtenshaw perhaps the record position is a very inconvenient message, could return the actual record along with it? Additionally, I think it would be nice to update the `__repr__` to avoid showing the entire vector on printing.

Reply:
If you agree, I will create some issues.

Reply:
I think it makes a big difference if record 0 or any other record raises the error in debugging. That said, these error messages could be improved with a representation of the record.

Reply:
@burtenshaw, I agree with showing the index but I think being able to see the record would be nice too.